docker/Dockerfile

```diff
@@ -101,7 +121,8 @@ RUN if [ -n "${CVE_UPDATES}" ]; then \
         rm -rf /var/cache/yum/*; \
     fi
 
-RUN useradd gpu-operator
-USER gpu-operator
+# The validator needs to run as uid 0 but the operator does not.
+# When deploying the operator, run as a non-root user.
+USER 0:0
```

Comment:
This is a bit strange since we're setting this here by the default command / entrypoint is `gpu-operator` and not `nvidia-validator`.

Reply:
I guess I could update the entrypoint to `nvidia-validator` for consistency. The reason I chose to default to `USER 0:0` is because it reduces the amount of changes required in our manifests -- we have many instances of the "validator" container, but only one "operator" container.

Reply:
Can these not be applied by the operator itself when construcing the validator?

Reply:
Updated to `65532:65532`. Also updated the code so that `runAsUser: 0` gets set in the securityContext for all validation containers we render.